tools/swift-inspect/Sources/SwiftInspectHeapWalk/SwiftInspectHeapWalk.cpp

```diff
@@ -0,0 +1,184 @@
+#if defined(_WIN32)
+
+#pragma comment(lib, "swiftCore.lib")
+
+#include <windows.h>
+#include <stdio.h>
+
+namespace {
+struct HeapEntry {
+  uintptr_t address;
+  uintptr_t size;
+};
+} // anonymous namespace
+
+#define BUF_SIZE 512
+#define BUF_NUM_ENTRIES (BUF_SIZE / sizeof(HeapEntry))
+#define SHARED_MEM_NAME "Local\\SwiftInspectFileMapping"
+#define READ_EVENT_NAME "Local\\SwiftInspectReadEvent"
+#define WRITE_EVENT_NAME "Local\\SwiftInspectWriteEvent"
```

Comment:
Do we need to worry about collisions in case of multiple running entries?  Perhaps we should use `.{ProcessID}` to disambiguate?